internal/tools/orchestrator/hepa/apipolicy/policies/ip/policy.go

```diff
@@ -220,16 +267,72 @@ func setMseIngressAnnotations(policyDto *PolicyDto) *apipolicy.IngressAnnotation
 	return ingressAnnotations
 }
 
-func (policy Policy) buildPluginReq(dto *PolicyDto) *providerDto.PluginReqDto {
+func (policy Policy) buildMSEPluginReq(dto *PolicyDto, zoneName string) *providerDto.PluginReqDto {
 	req := &providerDto.PluginReqDto{
-		Name:    mseCommon.MsePluginIP,
-		Config:  map[string]interface{}{},
-		Enabled: &dto.Switch,
+		Name:     mseCommon.MsePluginIP,
+		Config:   map[string]interface{}{},
+		Enabled:  &dto.Switch,
+		ZoneName: zoneName,
+	}
+
+	switch dto.IpSource {
+	case X_REAL_IP:
+		req.Config[mseCommon.MseErdaIpIpSource] = mseCommon.MseErdaIpSourceXRealIP
+	case X_FORWARDED_FOR:
+		req.Config[mseCommon.MseErdaIpIpSource] = mseCommon.MseErdaIpSourceXForwardedFor
+	default:
+		req.Config[mseCommon.MseErdaIpIpSource] = mseCommon.MseErdaIpSourceRemoteIP
+	}
+
+	req.Config[mseCommon.MseErdaIpAclType] = string(dto.IpAclType)
+	req.Config[mseCommon.MseErdaIpAclList] = dto.IpAclList
+
+	if dto.Switch {
+		req.Config[mseCommon.MseErdaIpRouteSwitch] = true
+	} else {
+		req.Config[mseCommon.MseErdaIpRouteSwitch] = false
 	}
 
 	return req
 }
 
+// 初创路由或者关闭路由策略（PolicyDto.Switch == false）的时候,都会进入 ParseConfig 的同一段逻辑中， 但:
+// 1. 如果是关闭路由策略，则对应的逻辑里需要清除已经配置的插件策略，\
+// 2. 如果是新建路由，实际上是不需要进行处理的，但可以延时等待拿到对应的新的路由信息，然后进行类似清除路由对应的策略配置的设置即可，但这个过程不能同步等待，因此异步执行，最多重试4次
+func (policy Policy) nonSwitchUpdateMSEPluginConfig(mseAdapter gateway_providers.GatewayAdapter, policyDto *PolicyDto, zoneName string) {
+	for i := 0; i < 3; i++ {
+		time.Sleep(10 * time.Second)
+		resp, err := mseAdapter.CreateOrUpdatePluginById(policy.buildMSEPluginReq(policyDto, strings.ToLower(zoneName)))
+		if err != nil {
+			if i == 2 {
+				logrus.Errorf("can not update mse erda-ip plugin for 4 times in 30s, err: %v", err)
+				return
+			}
+			continue
```

Comment:
```suggestion
func (policy Policy) nonSwitchUpdateMSEPluginConfig(mseAdapter gateway_providers.GatewayAdapter, policyDto *PolicyDto, zoneName string) {
	for i := 0; i < 4; i++ {
		resp, err := mseAdapter.CreateOrUpdatePluginById(policy.buildMSEPluginReq(policyDto, strings.ToLower(zoneName)))
		if err != nil {
			if i == 3 {
				logrus.Errorf("can not update mse erda-ip plugin for 4 times in 30s, err: %v", err)
				return
				}
			time.Sleep(10 * time.Second)
			continue
```

1. sleep 10s 和 continue 放一起更好吧，那如果一次成功的话，就不需要 sleep 10s
2. 得调整下 for 循环的值，i < 4; i == 3 才表示尝试了 4 次吧

Reply:
1. 进来之前已经做了一次，不成功才会进到这个地方，所以进来就要等 10s
2. 加上进来之前已经做了一次，所以  i == 2 表示 总共做了 4 次，描述有点问题

Reply:
OK